### PR TITLE
[node@4] accept string in zlib input

### PR DESCRIPTION
**Pull request by marcghorayeb**
Essentially allows strings to be used as input in place of buffers for zlib convenience methods.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://nodejs.org/docs/latest-v4.x/api/zlib.html#zlib_convenience_methods
- [x] Increase the version number in the header if appropriate. - not needed
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. - not needed

### CODE REVIEW

File: types/node/v4/index.d.ts
Comment (mhegazy) [inline types/node/v4/index.d.ts]: I would rather you keep this inlined as `Buffer | string`. it is easier for users to understand when looking at the signature help in their IDE, rather than having to go search for what `ZlibInput` really mean.

**Review by mhegazy — CHANGES_REQUESTED** [review]
this change needs to be done to all other node versions i.e. types/node/v*/index.d.ts, as well as latest (types/node/index.d.ts).

**Comment by mhegazy** [comment]
ping @kimushu  @ikokostya @Tyriar @ryota-ka @JoshGlazebrook @blakeembrey @bcherny @andy-ms @Loghorn @hinell

**Review by ghost — APPROVED** [review]

File: types/node/v4/index.d.ts
Comment (hinell) [inline types/node/v4/index.d.ts]: Agree.

**Comment by hinell** [comment]
>this change needs to be done to all other node versions

@mhegazy Would suggest to accept this PR and open a separate one to avoid overcrowding and closing due to negligence.

File: types/node/v4/index.d.ts
Comment (ikokostya) [inline types/node/v4/index.d.ts]: This change is needed? Type definition has *many* contributors and I don't think that all of them should be listed here.

File: types/node/v4/node-tests.ts
Comment (ikokostya) [inline types/node/v4/node-tests.ts]: 1. All tests are separated by modules. This test related to `zlib` module and for consistency should have comment 
  
   ```
   ////////////////////////////////////////////////////
   /// zlib tests : http://nodejs.org/api/zlib.html ///
   ////////////////////////////////////////////////////
   ```

2. Also for consistency maybe better use `namespace` instead function:

   ```ts
   namespace zlib_tests {
       const gzipped = zlib.gzipSync('test');
       const unzipped = zlib.gunzipSync(gzipped);
   }
   ```

File: types/node/v4/index.d.ts
Comment (hinell) [inline types/node/v4/index.d.ts]: I think everybody who have contributed to the module should be listed in or none of them.

Comment (ikokostya) [inline types/node/v4/index.d.ts]: I vote for `none`, because readers are first interested in the definitions, not in the list of contributors. 
`git` already keeps this information.

Comment (andy-ms) [inline types/node/v4/index.d.ts]: The `Definitions by:` list is used to determine who gets notified when changes are made to this file. The first two "authors" here can't receive notifications though. It would be good to get some people who would actually respond to new `@types/node` PRs.

**Comment by marcghorayeb** [comment]
Thanks for the feedback, I'll address them today!

File: types/node/v4/index.d.ts
Comment (marcghorayeb) [inline types/node/v4/index.d.ts]: As you wish for the contributors, I was just following the guideline in the readme that said that you could add yourself there. First time I'm contributing here :)

File: types/node/v4/index.d.ts
Comment (marcghorayeb) [inline types/node/v4/index.d.ts]: True, I didn't think of that. Will change this.

**Comment by marcghorayeb** [comment]
I've updated the PR, let me know if it's ok. I have another one coming up after this to homogenize the other versions as @mhegazy suggested.

**Review by ikokostya — APPROVED** [review]

File: types/node/v4/index.d.ts
Comment (andy-ms) [inline types/node/v4/index.d.ts]: `git blame` would include a lot more people than are really willing to help maintain the definitions.

**Review by hinell — APPROVED** [review]
LGTM

**Comment by marcghorayeb** [comment]
@mhegazy anything holding this back? once this is merged I can open the other PR for other versions of node 👍

**Pull request merged**
